mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -192,41 +293,159 @@ impl TendermintCoin {
 
         Ok(TendermintCoin(Arc::new(TendermintCoinImpl {
             ticker,
-            rpc_client,
+            rpc_clients,
             account_id,
             account_prefix: protocol_info.account_prefix,
             priv_key: priv_key.to_vec(),
             decimals: protocol_info.decimals,
             denom,
             chain_id,
+            gas_price: protocol_info.gas_price,
+            avg_block_time,
             sequence_lock: AsyncMutex::new(()),
+            tokens_info: Mutex::new(HashMap::new()),
         })))
     }
 
-    async fn my_account_info(&self) -> MmResult<BaseAccount, TendermintCoinRpcError> {
-        let path = AbciPath::from_str("/cosmos.auth.v1beta1.Query/Account").expect("valid path");
+    // TODO
+    // Save one working client to the coin context, only try others once it doesn't
+    // work anymore.
+    // Also, try couple times more on health check errors.
+    async fn rpc_client(&self) -> MmResult<HttpClient, TendermintCoinRpcError> {
+        for rpc_client in self.rpc_clients.iter() {
+            match rpc_client.perform(HealthRequest).timeout(Duration::from_secs(3)).await {
+                Ok(res) => match res {
+                    Ok(_) => return Ok(rpc_client.clone()),
+                    Err(e) => {
+                        log::warn!(
+                            "Recieved error from Tendermint rpc node during health check. Error: {:?}",
+                            e
+                        );
+                    },
+                },
+                Err(_) => {
+                    log::warn!("Tendermint rpc node: {:?} got timeout during health check", rpc_client);
+                },
+            };
```

Comment:
This can be rewrite as:
```rust
match rpc_client.perform(HealthRequest).timeout(Duration::from_secs(3)).await {
                Ok(Ok(_)) => return Ok(rpc_client.clone()),
                Ok(Err(e)) => log::warn!("Recieved error from Tendermint rpc node during health check. Error: {e:?}",),
                Err(_timeout) => log::warn!("Tendermint rpc node: {rpc_client:?} got timeout during health check"),
            };
```

Reply:
Seems much better thanks!